js/src/components/shipping-rate-section/estimated-shipping-rates-card/estimated-shipping-rates-card.js

```diff
@@ -28,93 +34,196 @@ const defaultShippingRate = {
  * with an UI, that allows to aggregate countries with the same rate.
  *
  * @param {Object} props
- * @param {Array<ShippingRateFromServerSide>} props.value Array of individual shipping rates to be used as the initial values of the form.
+ * @param {Array<ShippingRate>} props.value Array of individual shipping rates to be used as the initial values of the form.
  * @param {Array<CountryCode>} props.audienceCountries Array of country codes of all audience countries.
- * @param {(newValue: Object) => void} props.onChange Callback called with new data once shipping rates are changed.
+ * @param {(newValue: Array<ShippingRate>) => void} props.onChange Callback called with new data once shipping rates are changed.
  */
 export default function EstimatedShippingRatesCard( {
-	value: shippingRates,
 	audienceCountries,
+	value,
 	onChange,
 } ) {
 	const { code: currencyCode } = useStoreCurrency();
-	const actualCountryCount = shippingRates.length;
-	const actualCountries = new Map(
-		shippingRates.map( ( rate ) => [ rate.country, rate ] )
-	);
-	const remainingCountries = audienceCountries.filter(
-		( el ) => ! actualCountries.has( el )
-	);
-	const remainingCount = remainingCountries.length;
-
-	// Group countries with the same rate.
-	const countriesPriceArray = groupShippingRatesByMethodPriceCurrency(
-		shippingRates
-	);
+	const groups = groupShippingRatesByMethodCurrencyRate( value );
 
-	// Prefill to-be-added price.
-	if ( countriesPriceArray.length === 0 ) {
-		countriesPriceArray.push( {
-			countries: audienceCountries,
-			method: SHIPPING_RATE_METHOD.FLAT_RATE,
-			price: null,
-			currency: currencyCode,
-		} );
-	}
-
-	// Given the limitations of `<Form>` component we can communicate up only onChange.
-	// Therefore we loose the infromation whether it was add, change, delete.
-	// In autosave/setup MC case, we would have to either re-calculate to deduct that information,
-	// or fix that in `<Form>` component.
-	function handleDelete( deletedCountries ) {
-		onChange(
-			shippingRates.filter(
-				( rate ) => ! deletedCountries.includes( rate.country )
-			)
-		);
-	}
-	function handleAdd( { countries, currency, rate } ) {
-		// Split aggregated rate, to individial rates per country.
-		const addedIndividualRates = countries.map( ( country ) => ( {
+	/**
+	 * Event handler for adding new shipping rate group.
+	 *
+	 * Shipping rate group will be converted into shipping rates, and propagate up via `onChange`.
+	 *
+	 * @param {ShippingRateGroup} newGroup Shipping rate group.
+	 */
+	const handleAddSubmit = ( { countries, method, currency, rate } ) => {
+		const newShippingRates = countries.map( ( country ) => ( {
 			...defaultShippingRate,
 			country,
+			method,
```

Comment:
That's because I'm treating the components like a "black box", without looking into its implementation. We pass down the `method` value via props and get the value back via event handlers. The components _might_ change the value, and we should use that. (We only know that the components don't change the `method` value because we look into its implementation - a "white box".)

Side note: We probably don't need to be too concerned on this, because the `method` property shall be removed from the API side (see https://github.com/woocommerce/google-listings-and-ads/pull/1422#issuecomment-1100022747) and we would remove it from our UI code here soon.